### PR TITLE
CEDS-1411: last of the mappings in GoodsShipment Builders

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/services/mapping/goodsshipment/DestinationBuilder.scala [hunk 1]
```diff
@@ -17,29 +17,31 @@
...
+
+}
+
 object DestinationBuilder {
 
   def build(implicit cacheMap: CacheMap, choice: Choice): GoodsShipment.Destination =
     cacheMap
       .getEntry[DestinationCountries](DestinationCountries.formId)
-      .filter(data => isDefined(data.countryOfDestination))
+      .filter(_.countryOfDestination.nonEmpty)
```
Comment (ed0906) [inline app/services/mapping/goodsshipment/DestinationBuilder.scala]: 👍

File: app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala [hunk 2]
```diff
@@ -37,21 +44,42 @@ class GoodsShipmentBuilder @Inject()(
...
+
+    exportsCacheModel.locations.warehouseIdentification
+      .foreach(warehouseIdentification => warehouseBuilder.buildThenAdd(warehouseIdentification, goodsShipment))
+
+    exportsCacheModel.previousDocuments.foreach(
+      previousDocuments => previousDocumentsBuilder.buildThenAdd(previousDocuments, goodsShipment)
+    )
+
+    exportsCacheModel.parties.declarationAdditionalActorsData.foreach { declarationAdditionalActorsData =>
+      {
```
Comment (bambuchaAdm) [inline app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala]: Would you like to explain purpose of this scope ?

File: test/services/cache/ExportsCacheItemBuilder.scala [hunk 3]
```diff
@@ -83,23 +83,29 @@ trait ExportsCacheItemBuilder {
...
+        )
+    )
 
   def withDocumentsProduced(first: DocumentsProduced, docs: DocumentsProduced*): CachedItemModifier = cache => {
     val existing = cache.documentsProducedData.map(_.documents).getOrElse(Seq.empty)
     cache.copy(documentsProducedData = Some(DocumentsProducedData(existing ++ Seq(first) ++ docs)))
   }
 
-  def withDocumentsProduced(docs: DocumentsProducedData): CachedItemModifier =
+  def withDocumentsProducedData(docs: DocumentsProducedData): CachedItemModifier =
```
Comment (ed0906) [inline test/services/cache/ExportsCacheItemBuilder.scala]: Minor but *Data* just adds clutter to the method name, not sure why it's been added back on

File: test/services/mapping/goodsshipment/GoodsShipmentBuilderSpec.scala [hunk 4]
```diff
@@ -24,32 +24,50 @@ import forms.declaration.DestinationCountriesSpec.correctDestinationCountries
...
 import services.cache.{ExportItem, ExportsCacheModelBuilder}
+import services.mapping.goodsshipment.consignment.ConsignmentBuilder
 import services.mapping.governmentagencygoodsitem.GovernmentAgencyGoodsItemBuilder
 import wco.datamodel.wco.dec_dms._2.Declaration
 
 class GoodsShipmentBuilderSpec extends WordSpec with Matchers with ExportsCacheModelBuilder with MockitoSugar {
 
-  private val goodsShipmentNatureOfTransactionBuilder = mock[GoodsShipmentNatureOfTransactionBuilder]
-  private val consigneeBuilder = mock[ConsigneeBuilder]
+  private val mockGoodsShipmentNatureOfTransactionBuilder = mock[GoodsShipmentNatureOfTransactionBuilder]
```
Comment (ed0906) [inline test/services/mapping/goodsshipment/GoodsShipmentBuilderSpec.scala]: minor again but `mock` just adds clutter to the variable names

File: app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala (hunk 2 above)
Comment (ed0906) [reply app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala]: I think what he means is you probably don't need the `{` and `}`

File: test/services/cache/ExportsCacheItemBuilder.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply test/services/cache/ExportsCacheItemBuilder.scala]: As it described what it is doing. there are two object DocumentsProduced and documentsProducedData... its adding the one with data...

File: test/services/mapping/goodsshipment/GoodsShipmentBuilderSpec.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply test/services/mapping/goodsshipment/GoodsShipmentBuilderSpec.scala]: don think this matters...its a mock... call it a mock

File: app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala]: It reads better for me as you can see the scope as you have nested foreach. But don't have strong feelings about it either way.

**Pull request merged**